src/command/restore/restore.c

```diff
@@ -1368,14 +1392,24 @@ restoreSelectiveExpression(Manifest *manifest)
                 if (strLstExists(systemDbIdList, includeDb))
                     THROW(DbInvalidError, "system databases (template0, postgres, etc.) are included by default");
 
+                // Error if the db id is in the exclude list
+                if (strLstExists(excludeDbIdList, includeDb))
+                    THROW_FMT(DbInvalidError, "database to include '%s' is in the exclude list", strZ(includeDb));
+
                 // Remove from list of DBs to zero
                 strLstRemove(dbList, includeDb);
             }
 
-            // Exclude the system databases from the list
+            // Remove the system databases from list of DBs to zero unless they are excluded explicitly
             strLstSort(systemDbIdList, sortOrderAsc);
+            strLstSort(excludeDbIdList, sortOrderAsc);
+            systemDbIdList = strLstMergeAnti(systemDbIdList, excludeDbIdList);
             dbList = strLstMergeAnti(dbList, systemDbIdList);
 
+            // Only exclude specified db in case no db to include has been provided
+            if (strLstSize(includeList) <= 0)
+                dbList = strLstDup(excludeDbIdList);
```

Comment:
This and the above statements might be better as an IF-ELSE - since why manipulate the dbList if it's just going to be overwritten, right? Also should use strLstEmpty vs strLstSize. So maybe:

```
// Only exclude specified db in case no db to include has been provided
if (strLstEmpty(includeList))
{
    dbList = strLstDup(excludeDbIdList);
}
// Else, remove the system databases from list of DBs to zero unless they are excluded explicitly
else
{
    strLstSort(systemDbIdList, sortOrderAsc);
    systemDbIdList = strLstMergeAnti(systemDbIdList, excludeDbIdList);
    dbList = strLstMergeAnti(dbList, systemDbIdList);
}
```

Reply:
indeed, more readable too with that style. Fixed in f46f1c3 :-)